### PR TITLE
lib/model: Allow empty subdirs in scan request (fixes #3829)

### LINKED ISSUE #3829: Receiving 'invalid subpath' errors on syncthing-inotify full folder sync trigger
I have been making some large scale file moves in one of my synced folders and this has caused syncthing to go **FAILED** indicating an **invalid subpath** error. 

### Current Behavior
- `syncthing-inotify` triggers scan of full sync path
- `syncthing` returns **invalid subpath** both to the `syncthing-inotify` request and shows same in the web UI

Looking at the logs for syncthing-inotify, I see the following:
```
[OK] Watching vc: /home/mikej/vc
[WARNING] Failed to perform request Post http://127.0.0.1:8384/rest/db/scan?folder=vc&next=3600&sub=.stfolder: net/http: request canceled (Client.Timeout exceeded while awaiting headers)
[WARNING] Request to delay scanning of vc failed
[WARNING] http://127.0.0.1:8384/rest/db/scan?folder=vc&next=3600&sub=
[WARNING] Error: Status 500 != 200 for POST: vc, invalid subpath
[WARNING] Syncthing failed to index changes for  vc Invalid HTTP status code
[WARNING] http://127.0.0.1:8384/rest/db/scan?folder=vc&next=3600&sub=
[WARNING] Error: Status 500 != 200 for POST: vc, invalid subpath
[WARNING] Syncthing failed to index changes for  vc Invalid HTTP status code
...
```

### Expected Behavior
- `syncthing-inotify` triggers scan of full sync path
- `syncthing` completes the scan successfully.

I will also note that if I stop/disable `syncthing-inotify` and hit **Rescan** in the web UI, the sync completes successfully with no errors.

I checked at the recent issues and though I believe it may be the _result_ of #3805 - because the timing works and #3805 made adjustments to the subpath detection - the problem behavior is distinct from that issue. I unfortunately could not work out how to write a test that demonstrated the error (still very new to Go), so I will have to defer that to the syncthing devs.

### Version Information

Syncthing Version: v0.14.15
Syncthing-inotify Version: v0.8.4
OS Version: Arch Linux 64bit
Browser Version: N/A

### PR DESCRIPTION
**Pull request by calmh**
### Purpose

Pesky invalid subpaths

### CODE REVIEW

File: lib/model/model_test.go [hunk 1]
```diff
@@ -2181,6 +2181,21 @@ func TestIssue3804(t *testing.T) {
...
+	m.AddFolder(defaultFolderConfig)
+	m.StartFolder("default")
+	m.ServeBackground()
+	defer m.Stop()
+
+	// Empty subdirs should be accepted
+
+	if err := m.ScanFolderSubdirs("default", []string{""}); err != nil {
+		t.Error("Unexpected error:", err)
+	}
```
Comment (AudriusButkevicius) [inline lib/model/model_test.go]: What happens with `[]string{"", "/foo"}`?

Comment (calmh) [inline lib/model/model_test.go]: Eh. Yeah, `""` gets filtered out and we're left with just `"foo"` which is wrong. Good catch.

**Comment by AudriusButkevicius** [comment]
@st-review merge

**Comment by st-review** [comment]
@AudriusButkevicius: Build status is `pending`. I'll wait until it goes green and then merge!

**Pull request closed without merging**